src/target/riscv/riscv.c

```diff
@@ -5287,6 +5304,14 @@ static const struct command_registration riscv_exec_command_handlers[] = {
 		.usage = "[('eq'|'napot'|'ge_lt'|'all') ('wp'|'none')]",
 		.help = "Control whether OpenOCD is allowed to use certain RISC-V trigger features for watchpoints."
 	},
+	{
+		.name = "reserve_trigger",
+		.handler = handle_reserve_trigger,
+		/* TODO: Move this to COMMAND_ANY */
```

Comment:
Is there a reason this has not been done yet?

Reply:
To reserve a trigger when a trigger module is not available or when the number of available triggers is not enough for the given index to be valid seems like an error.
Therefore, `handle_reserve_trigger` calls `riscv_enumerate_triggers()` which is only available after `examine()` (i.e. after TCL's `init` is called, i.e. in `COMMAND_EXEC` mode).

This can be mitigated by adding RISC-V specific target configure options that will allow the user to specify the number of available triggers in `COMMAND_CONFIG` mode, making `examine` just check that the configured parameter is correct (or even do nothing at all).